libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -589,8 +587,8 @@ void CodeTransform::operator()(FunctionDefinition const& _function)
 	else
 		m_assembly.appendJump(stackHeightBefore - _function.returnVariables.size());
 	m_stackAdjustment -= localStackAdjustment;
-	m_assembly.appendLabel(afterFunction);
 	checkStackHeight(&_function);
+	m_assembly.setStackHeight(stackHeightBefore);
```

Comment:
What happens if you move this one line up?

Reply:
Nothing. I get the same 2 test failures.